### PR TITLE
Prefer optional non-empty strings

### PR DESCRIPTION
**Pull request by smortex**
#### Pull Request (PR) description

Instead of using empty strings for optional values, change the data type
to Optional[String[1]] to ensure that either a valid String is provided
or no value is passwd.

A side effect is that these parameters cannot be set explicitely undef
by the end user if they have a default value, and $poudriere::zpool is
concerned by this.  But since this default value is bogus, change it to
undef to avoid the issue.

#### This Pull Request (PR) fixes the following issues

Fixes parts of #52

### CODE REVIEW

**Review by kenyon — APPROVED** [review]

**Review by xaque208 — APPROVED** [review]
:+1:

File: templates/poudriere.conf.erb [hunk 1]
```diff
@@ -10,20 +10,20 @@
 # poudriere.
 #
 #ZPOOL=
-<% if @zpool and @zpool!='' -%>
+<% if @zpool -%>
```
Comment (ekohl) [inline templates/poudriere.conf.erb:13]: Does it make sense to switch over to epp and use `unless empty($zpool)`?

Comment (smortex) [reply templates/poudriere.conf.erb:13]: I added switching from ERB to EPP in the modernization and cleanup issue

**Comment by smortex** [comment]
:eyes: I have a few things I would like to improve but they depend on this…  I think it is ready for merging, are there any issues?

**Comment by ghoneycutt** [comment]
Added backwards-incompatible because it changes the default value for `$zpool`

**Review by ghoneycutt — APPROVED** [review]

**Pull request merged**